ee/tabby-ui/app/search/components/search.tsx

```diff
@@ -120,79 +136,156 @@ export const SearchContext = createContext<SearchContextValue>(
+        startCursor
+        endCursor
+      }
+    }
+  }
+`)
+
 export function Search() {
+  const { updateUrlComponents, pathname } = useRouterStuff()
+  const [activePathname, setActivePathname] = useState<string | undefined>()
+  const [isPathnameInitialized, setIsPathnameInitialized] = useState(false)
   const isChatEnabled = useIsChatEnabled()
   const [searchFlag] = useEnableSearch()
-  const [conversation, setConversation] = useState<ConversationMessage[]>([])
+  const [messages, setMessages] = useState<ConversationMessage[]>([])
   const [showStop, setShowStop] = useState(true)
   const [container, setContainer] = useState<HTMLDivElement | null>(null)
   const [title, setTitle] = useState('')
   const [isReady, setIsReady] = useState(false)
   const [extraContext, setExtraContext] = useState<AnswerEngineExtraContext>({})
-  const [currentLoadindId, setCurrentLoadingId] = useState<string>('')
+  const [currentUserMessageId, setCurrentUserMessageId] = useState<string>('')
+  const [currentAssistantMessageId, setCurrentAssistantMessageId] =
+    useState<string>('')
   const contentContainerRef = useRef<HTMLDivElement>(null)
   const [showSearchInput, setShowSearchInput] = useState(false)
   const [isShowDemoBanner] = useShowDemoBanner()
   const router = useRouter()
-  const initCheckRef = useRef(false)
+  const initializing = useRef(false)
   const { theme } = useCurrentTheme()
   const [devPanelOpen, setDevPanelOpen] = useState(false)
-  const [conversationIdForDev, setConversationIdForDev] = useState<
-    string | undefined
-  >()
+  const [messageIdForDev, setMessageIdForDev] = useState<string | undefined>()
   const devPanelRef = useRef<ImperativePanelHandle>(null)
   const [devPanelSize, setDevPanelSize] = useState(45)
   const prevDevPanelSize = useRef(devPanelSize)
 
+  const threadId = useMemo(() => {
+    const regex = /^\/search\/(.*)/
+    if (!activePathname) return undefined
+
+    return activePathname.match(regex)?.[1]
+  }, [activePathname])
+
   const [{ data }] = useQuery({
     query: repositoryListQuery
   })
   const repositoryList = data?.repositoryList
 
-  const { triggerRequest, isLoading, error, answer, stop } = useTabbyAnswer({
-    fetcher: tabbyFetcher
+  const [afterCursor, setAfterCursor] = useState<string | undefined>()
+  const [{ data: threadMessages, fetching: fetchingMessages }] = useQuery({
+    query: listThreadMessages,
+    variables: {
+      threadId: threadId as string,
+      first: 20,
+      after: afterCursor
+    },
+    pause: !threadId || isReady
+  })
+
+  // todo scroll and setAfterCursor
+  useEffect(() => {
+    if (threadMessages?.threadMessages?.edges?.length) {
+      const messages = threadMessages.threadMessages.edges
+        .map(o => o.node)
+        .slice()
+      setMessages(messages)
+      setIsReady(true)
+    }
+  }, [threadMessages])
+
+  const onThreadCreated = (threadId: string) => {
+    updateUrlComponents({
+      pathname: `/search/${threadId}`
```

Comment:
```suggestion
      pathname: `/search/${threadId}`
```

pathname shall be something like `/search/what-does-chat-completion-do-${threadId}`

Reply:
done